### DISCUSSION TITLE
External cameras?

### DISCUSSION THREAD

**Discussion by jf-64** [jomjol/AI-on-the-edge-device#3387] [Q&A]
Hi All,

The project is currently stuck to ESP hardware. Wouldn't make sense to generalize the use of the [workflow](https://github.com/jomjol/AI-on-the-edge-device?tab=readme-ov-file#workflow-), thus, take the image also from external cameras (beside the ESP one) and the software as a HACS module?

Regards

**Comment by caco3** [comment]
This could be done but is out of scope of this project.
We want to keep it simple and easy for novice users.

**Comment by jf-64** [comment]
I'm not quite sure, that the generalization would make it much more complicated, since the common workflows (e.g. for the current HW) could be automatically generated. Thus, for the novice users it would be transparent, but the computation for everyone much faster - with the ability to use also other _image sources_*...

*: I'm explicitly not talking about cameras only.

**Reply by caco3** [reply]
But it would be quite som effort to implement and document it., It also needs to be tested.
We are just a very small team of developers and our time is limited.
If you are willing to contribute your time and work for it, we can have a look.
